ui/src/app/views/resource-card.tsx

```diff
@@ -636,15 +639,25 @@ export class ResourceCard extends React.Component<Props, State> {
           oldName={this.props.resourceCard.dataSet.name}
           existingNames={this.props.existingNameList}/>
       }
-      {this.state.showDirtyDataSetModal && <Modal>
+      {this.state.dataSetByResourceIdList.length > 0 && <Modal>
         <ModalTitle>WARNING</ModalTitle>
         <ModalBody>
           <div style={{paddingBottom: '1rem'}}>
-            This {this.resourceType} is being used in a Data Set. Do you still want to delete?
+            The {this.resourceType} <b>{fp.startCase(this.displayName)}&nbsp;</b>
+            is referenced by the following Data Sets:
+            <b>
+              &nbsp;
+              {fp.join(' , ' ,
+                this.state.dataSetByResourceIdList.map((data) => data.name))}
+            </b>.
+            Deleting the {this.resourceType} <b>{fp.startCase(this.displayName)}</b>
+            will make these Data Sets
```

Comment:
The spacing on this text seems off, I'm surprised the linter is happy with this, but can we put this and the `unavailable for use` on the same line?